libmambapy/tests/test_solver_libsolv.py

```diff
@@ -193,31 +207,43 @@ def tmp_repodata_json(tmp_path):
     return file
 
 
-@pytest.mark.parametrize(
-    ["add_pip_as_python_dependency", "package_types", "repodata_parser"],
-    itertools.product(
-        [True, False],
-        ["TarBz2Only", "CondaOrElseTarBz2"],
-        ["Mamba", "Libsolv"],
-    ),
-)
+@pytest.mark.parametrize("add_pip_as_python_dependency", [True, False])
+@pytest.mark.parametrize("package_types", ["TarBz2Only", "CondaOrElseTarBz2"])
+@pytest.mark.parametrize("repodata_parser", ["Mamba", "Libsolv"])
+@pytest.mark.parametrize("matchspec_parser", ["Mixed", "Mamba", "Libsolv"])
 def test_Database_RepoInfo_from_repodata(
-    tmp_path, tmp_repodata_json, add_pip_as_python_dependency, package_types, repodata_parser
+    tmp_path,
+    tmp_repodata_json,
+    add_pip_as_python_dependency,
+    package_types,
+    repodata_parser,
+    matchspec_parser,
 ):
-    db = libsolv.Database(libmambapy.specs.ChannelResolveParams())
+    db = libsolv.Database(
+        libmambapy.specs.ChannelResolveParams(),
+        matchspec_parser=matchspec_parser,
+    )
 
     url = "https://repo.mamba.pm"
     channel_id = "conda-forge"
 
-    # Json
-    repo = db.add_repo_from_repodata_json(
-        path=tmp_repodata_json,
-        url=url,
-        channel_id=channel_id,
-        add_pip_as_python_dependency=add_pip_as_python_dependency,
-        package_types=package_types,
-        repodata_parser=repodata_parser,
-    )
+    def add_repo_json():
+        return db.add_repo_from_repodata_json(
+            path=tmp_repodata_json,
+            url=url,
+            channel_id=channel_id,
+            add_pip_as_python_dependency=add_pip_as_python_dependency,
+            package_types=package_types,
+            repodata_parser=repodata_parser,
+        )
+
+    if (repodata_parser == "Libsolv") and (matchspec_parser != "Libsolv"):
+        with pytest.raises(libmambapy.MambaNativeException) as e:
+            add_repo_json()
+            assert "Libsolv repodata parser uses only its own MatchSpec parser" in e
```

Comment:
How about a verbose message to explain the cause of the error to the end-user?
```suggestion
            assert "Libsolv repodata parser must only be used with its own MatchSpec parser" in e
            assert f"Currently {repodata_parser} is used for the repodata parser" in e
            assert f"while {matchspec_parser} is used for the MatchSpec parser" in e
```

Reply:
Not sure about which message. It's not a constraint that it cannot work, it just that there is nothing that allow us to plug a different parser there.

Reply:
I think my remark was not clear.

If I were an user, I do not think I would understand the cause of this error and what I should do to resolve it.

Reply:
Solved in the error message, but keeping the Python tests light since they should only test bindings are working properly.